File: DI/DI-Dagger-Java/src/main/java/DI/persona/modules/PersonasDBModule.java
package DI.persona.modules;

import dagger.Binds;
import dagger.Module;
import model.persona.services.IPersonaStorage;
import model.persona.services.PersonaDataBaseStorage;

import javax.inject.Singleton;

@Module
public interface PersonasDBModule {
    @Singleton
    @Binds
    IPersonaStorage bindPersonaDBStorage(PersonaDataBaseStorage impl);
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/DI/personas/modules/PersonasDBModule.kt
package DI.personas.modules

import dagger.Binds
import dagger.Module
import models.personas.services.IPersonaStorage
import models.personas.services.PersonaDataBaseStorage
import javax.inject.Singleton


@Module
interface PersonasDBModule {
    @Singleton
    @Binds
    fun bindPersonaDBStorage(impl: PersonaDataBaseStorage): IPersonaStorage
}
